src/back/CodeGen/Expr.hs

```diff
@@ -1062,11 +1054,41 @@ instance Translatable A.Expr (State Ctx.Context (CCode Lval, CCode Stat)) where
     else
       error $ "Expr.hs: Cannot translate forward of ''" ++ show expr ++ "'"
 
-  translate A.Forward{A.forwardExpr = A.FutureChain{}} =
-    error "Expr.hs: Forwarding of chaining not implemented"
-  translate A.Forward{A.forwardExpr} =
-    error $ "Expr.hs: Target of forward is not method call or future chain: '" ++
-            show forwardExpr ++ "'"
+  translate A.Forward{A.emeta, A.forwardExpr = fchain@A.FutureChain{A.future, A.chain}} = do
+    (nfuture,tfuture) <- translate future
+    (nchain, tchain)  <- translate chain
+    eCtx <- gets $ Ctx.getExecCtx
+    withForwarding <- gets Ctx.withForwarding
+    let ty = getRuntimeType chain
+        dtraceExit = head (getDtraceExit eCtx)
+    if withForwarding
```

Comment:
`withForwarding` is misleading name. My understanding is that you translate the code twice, one for when it is called asynchronously, and once for when it is called `synchronously`. The name `withForwarding` should reflect that.

Furthermore, the top-level names in the generated code should reflect that too.